src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/MLKemOpenSsl.NotSupported.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.Versioning;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Security.Cryptography
+{
+    [UnsupportedOSPlatform("android")]
+    [UnsupportedOSPlatform("browser")]
+    [UnsupportedOSPlatform("ios")]
+    [UnsupportedOSPlatform("osx")]
+    [UnsupportedOSPlatform("tvos")]
+    [UnsupportedOSPlatform("windows")]
+    [Experimental(Experimentals.PostQuantumCryptographyDiagId)]
+    public sealed class MLKemOpenSsl : MLKem
+    {
+        public MLKemOpenSsl(SafeEvpPKeyHandle pkeyHandle) : base(MLKemAlgorithm.MLKem512)
```

Comment:
Bleh.  We probably want to partial out this type so we have sanity on the doc comments, the experimental attribute, and any OS support attributes we end up with.

Reply:
You can't have a partial constructor, but agree it would help with the class and other members.